PhenotypeOrthologous.pm

```diff
@@ -0,0 +1,216 @@
+
+  my $file = $params->{file};  
+  my $model = $params->{model};
+
+  die "File needs to be specified to run the PhenotypeOrthologus plugin. \n" if  (!$file);
+  $self->add_file($file);
+
+  if ( defined($model) && $model ne "rattus_norvegicus" && $model ne "rat" && $model ne "mouse" && $model ne "mus_musculus") {
+    die "PhenotypeOrthologous plugin only works for rat and mouse \n"
+  }
+  
+  if (defined($model)) {
+    $model = "rat" if $model eq "rattus_norvegicus";
+    $model = "mouse" if $model eq "mus_musculus";
+  }
+  
+  $self->{model} = $model;
+  $self->{params} = $params;
+
+
+  return $self;
+
+}
+
+sub feature_types {
+  return ['Transcript'];
+}
+
+sub get_header_info {
+  my $self = shift;
+  
+  my %header;
+
+
+  if (keys(%{$self->{params}}) == 1)  {
+    $header{"PhenotypeOrthologous_RatOrthologous_geneid"} = "PhenotypeOrthologous RatGene associated with Rat";
+    $header{"PhenotypeOrthologous_RatOrthologous_phenotype"} = "PhenotypeOrthologous RatPhenotypes associated with orthologous genes in Rat";
+    $header{"PhenotypeOrthologous_MouseOrthologous_geneid"} = "PhenotypeOrthologous MouseGene associated with Mouse";
+    $header{"PhenotypeOrthologous_MouseOrthologous_phenotype"} = "PhenotypeOrthologous MousePhenotypes associated with orthologous genes in Mouse";
+  }
+
+  if (defined($self->{model}) && $self->{model} eq "rat" ) {
+    $header{"PhenotypeOrthologous_RatOrthologous_geneid"} = "PhenotypeOrthologous RatGene associated with Rat";
+    $header{"PhenotypeOrthologous_RatOrthologous_phenotype"} = "PhenotypeOrthologous RatPhenotypes associated with orthologous genes in Rat";
+  }
+
+  if (defined($self->{model}) && $self->{model} eq "mouse" ) {
+    $header{"PhenotypeOrthologous_MouseOrthologous_geneid"} = "PhenotypeOrthologous MouseGene associated with Mouse";
+    $header{"PhenotypeOrthologous_MouseOrthologous_phenotype"} = "PhenotypeOrthologous MousePhenotypes associated with orthologous genes in Mouse";
+  }
+
+  return \%header;
+
+}
+
+sub run {
+  my ($self, $tva) = @_;
+  my $vf = $tva->variation_feature;
+
+  my $transcript = $tva->transcript;
+  my $gene_id =  $transcript->{_gene}->stable_id;
+
+  my ($vf_start, $vf_end) = ($vf->{start}, $vf->{end});
+  ($vf_start, $vf_end) = ($vf_end, $vf_start) if ($vf_start > $vf_end);
+
+  my ($res) = grep {
+    $_->{gene_id} eq $gene_id
+  } @{$self->get_data($vf->{chr}, $vf_start, $vf_end)};
+ 
+  return $res ? $res->{result} : {};
+
+}
+
+
+sub parse_data {
+  my ($self, $line) = @_;
+
+  my ($c, $grc, $feat, $s, $e, $n, $str, $n2, $note) = split /\t/, $line;
+
+  my @fields = split /;/, $note;
+  my %data_fields;
+  
+  foreach my $field (@fields){
+    my ($key, $value) = split /=/, $field;
+    $data_fields{$key} = $value;
+  }
+  
+  if (!$self->{model}) {
+    return {
+      start => $s,
+      end => $e,
+      gene_id => $data_fields{"gene_id"},
+      result => {
+        PhenotypeOrthologous_RatOrthologous_geneid => $data_fields{"Rat_gene_id"},
```

Comment:
We need to test if each of these exist and not return empty strings for mouse if only rat data exists.